### PR TITLE
Add viscous penalty to the edge scheme symmetry bc

### PR DESCRIPTION
**Pull request by psakievich**
**Pull-request type:**
- [ ] Bug fix 
- [ ] Documentation update
- [x] Feature enhancement

## Description of the pull-request

Add the same penalty factor @rcknaus added for #616 to the edge scheme.  Also create an option to control this penalty factor via solution options

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [x] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [ ] Linux
    - [x] MacOS
  - Compilers 
    - [ ] GCC
    - [x] LLVM/Clang
    - [ ] Intel compilers
    - [ ] NVIDIA CUDA
- [x] Compiles without warnings
- [x] Passes all unit tests
- [ ] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [ ] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [ ] New regression tests exercising the new code

### CODE REVIEW

**Comment by psakievich** [comment]
@rcknaus this change is changing the gold matrix in `UnitTestMomentumSymmetryEdge.C`.  What is the scenario we are using to set up the matrix for this? We can either:

1. regenerate the matrix
2. set the penalty factor to 0 for the test
3. set the default penalty factor to 0

I'm in favor of 1 or 2. 2 is definitely the easiest, but 1 could be considered more proper?  thoughts?

File: include/SolutionOptions.h [hunk 1]
```diff
@@ -150,6 +150,7 @@ class SolutionOptions
   double earthAngularVelocity_;
   double latitude_;
   double raBoussinesqTimeScale_;
+  double symmetryBcPenatlyFactor_;
```
Comment (sayerhs) [inline include/SolutionOptions.h:153]: Typo in `penalty`

**Comment by sayerhs** [comment]
@psakievich I would prefer the original unit test retain `penaltyFactor_ = 0.0` and then add a new unit test that tests the intended penaltyFactor_.

**Comment by psakievich** [comment]
@sayerhs that is a good suggestion.

**Comment by psakievich** [comment]
Going to run the regressions tests this morning.  I expect some level of diffs to crop up in the edge tests using the symmetry bc similar to the diffs created by #616.

**Comment by sayerhs** [comment]
@psakievich I feel that setting the default value to zero would be least surprising to users who reuse their old input files and reproduce runs. Without any change in the input file they might see slight changes in solutions that could lead to confusion if they are not closely tracking nalu-wind commits or catch this change in default value. 

Is there a significant improvement in solution at the boundary if we switch the default to 2.0 for flat boundaries? That would be an argument to break backwards compatibility. In this situation, we might want to print a warning message if the user hasn't explicitly set penalty factor in the input file alerting them that the default behavior has changed.

As for regression tests, I don't see an issue with re-blessing if they are updated to use the new penalty factor of 2.0.

**Comment by psakievich** [comment]
@sayerhs from the initial studies @rcknaus and I have done the flat boundaries behave like strong symmetry is applied with the penalty factor of 2.0.  So yes there is an improvement.  But the more I think about it, the more I think we should just leave it at 0.0 for the default for now.  We have only done a handful of tests so I can't really speak to full ramifications at this point.  Adding the penalty should also increase the error to some degree in the interior so my vote is to make this a beta capability for now and keep the current default.

**Comment by sayerhs** [comment]
@psakievich Leaving the default to 0.0 is good with me. I suppose you will add or modify a regression test to handle exercising `penaltyFactor  = 2.0` case?

**Comment by psakievich** [comment]
@sayerhs sure I can do that as well.

**Comment by psakievich** [comment]
@sayerhs @rcknaus I've removed the WIP flag and I believe this is ready for merging. I ran a subset of the regression tests and the unit tests and didn't see any failures on my mac.  I believe some diffs will occur in a few element scheme tests since @rcknaus turned on the penalty factor for that scheme by default but they should be small. I wasn't able to detect them in my local testing. When we identify them we can set the penalty factor for them in their input decks or just rebless.

**Review by sayerhs — APPROVED** [review]

**Pull request merged**